packages/next-app/src/app/internalmanager/(components)/TokenSelect.tsx

```diff
@@ -96,27 +99,31 @@ function TokenModal({
 
   const internalBalancesTokenAdresses = internalBalanceData?.user
     ?.userInternalBalances
-    ? internalBalanceData.user.userInternalBalances.map(
-        (token) => token.tokenInfo.address
+    ? internalBalanceData.user.userInternalBalances.map((token) =>
+        token.tokenInfo.address.toLowerCase()
       )
     : [];
 
-  async function getWalletBalance(
-    internalBalancesTokenAdresses: `0x${string}`[]
-  ) {
+  const tokenListAdresses = tokenList
+    ? tokenList.map((token) => token.address.toLowerCase())
+    : [];
+
+  const tokenAdresses = [
+    ...new Set([...internalBalancesTokenAdresses, ...tokenListAdresses]),
+  ];
+
+  async function getWalletBalance(tokenAdresses: `0x${string}`[]) {
     const walletBalanceData: TokenWalletBalance[] = [];
-    const walletBalancePromises = internalBalancesTokenAdresses.map(
-      async (tokenAddress) => {
-        const tokenData = await fetchBalance({
-          address: addressLower as `0x${string}`,
-          token: tokenAddress,
-        });
-        walletBalanceData.push({
-          ...tokenData,
-          tokenAddress: tokenAddress,
-        });
-      }
-    );
+    const walletBalancePromises = tokenAdresses.map(async (tokenAddress) => {
+      const tokenData = await fetchBalance({
```

Comment:
wondering if there's a batch function for this. it'd be better to do a multicall and get balances for all tokens instead of doing it iteratively.

Reply:
yup, tried to search for it on wagmi last week but found nothing. We can try to find it tomorrow